### PR TITLE
DHSCFT-775: convert to csv utility function

### PR DESCRIPTION
**Pull request by stuc-bjss**
# Description

Jira ticket: [DHSCFT-775](https://bjss-enterprise.atlassian.net/browse/DHSCFT-775)

- pre-requisite for the export to csv functionality
- trying not to add external library for something fairly simple

## Changes

- Added convert data arrays to csv string

## Validation

- Added unit tests passing incorrect types and extra commas, quotes, line breaks etc

### CODE REVIEW

File: frontend/fingertips-frontend/lib/downloadHelpers/convertToCsv.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  const needsQuotes = /[",]/.test(fieldString);
+  const escapedField = fieldString.replace(/"/g, '""');
+  return needsQuotes ? `"${escapedField}"` : escapedField;
+};
+
+export const convertRowToCsv = (row: CsvField[]): string =>
+  row.map(convertFieldToCsv).join(delimiter);
+
+export const convertToCsv = (data: CsvField[][]): string =>
+  data.map(convertRowToCsv).join(lineBreak);
```
Comment (williamwillis11) [inline frontend/fingertips-frontend/lib/downloadHelpers/convertToCsv.ts:19]: do we want to handle (and add a test for) if data is passed to this function as an empty array?

**Review by Joe-Hendley — APPROVED** [review]
